src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -38,7 +40,8 @@ public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs
         this.addressBook = new AddressBook(addressBook);
         this.userPrefs = new UserPrefs(userPrefs);
         filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
-        filteredAppointments = new FilteredList<>(this.userPrefs.getAppointmentList());
+        appointmentList = new AppointmentList(this.userPrefs.getAppointmentList());
+        filteredAppointments = new FilteredList<>(appointmentList.asUnmodifiableObservableList());
     }
```

Comment:
Ah okay I see what you did here.

Reply:
very hackish but it'll do for now

Reply:
I needed a new class to handle ObservableList and a way to get data for use